File: scripts/init_readme.py
with open("./scripts/project.txt") as f:
    PROJECT_NAME = f.read().strip().split("PROJECT_NAME=")[1]

def main():
    local_dev = (
        "### Local Development\n\n"
        "To get this running locally do the following.\n\n"
        "1. Make sure you have Docker desktop installed.\n"
        "2. Make sure you have mprocs installed.\n"
        "3. To start both the front and back end(s), type `mprocs` in the terminal."
    )

    sections = [
        f"## {PROJECT_NAME.capitalize()}",
        "## Demo Video",
        "## Architecture",
        "## Tech Stack",
        "### Front-End",
        "### Back-End",
        "### Database",
        local_dev,
    ]

    README = "\n\n".join(sections) + "\n"

    with open("./README.md", "w") as f:
        f.write(README)

    print(f"README.md initialized for '{PROJECT_NAME}'")

if __name__ == "__main__":
    main()

File: scripts/init_readme.rb
def init_readme
  project_name = File.read('./scripts/project.txt').strip.split('PROJECT_NAME=').last

  local_dev = <<~TEXT.chomp
    ### Local Development

    To get this running locally do the following.

    1. Make sure you have Docker desktop installed.
    2. Make sure you have mprocs installed.
    3. To start both the front and back end(s), type `mprocs` in the terminal.
  TEXT

  sections = [
    "## #{project_name.capitalize}",
    '## Demo Video',
    '## Architecture',
    '## Tech Stack',
    '### Front-End',
    '### Back-End',
    '### Database',
    local_dev
  ]

  File.write('./README.md', sections.join("\n\n") + "\n")

  puts "README.md initialized for '#{project_name}'"
end
